README.md

```diff
@@ -1,2 +1,202 @@
+const { CBPromise } = require('@consento/promise/cbPromise')
+
+const p = new CBPromise()
+p.cb(new Error(), 'result') // traditional callback API
+p.resolve('result') // or use traditional resolve
+p.reject(new Error()) // and reject
+```
+
+This may come in handy when you need a callback but want to await
+the promise.
+
+```javascript
+async function test () {
+  const p = new CBPromise()
+  setTimeout(p.resolve, 500)
+  await p
+}
+```
+
+#### `checkpoint()`
+
+Allows the creation of a checkpoint function that aborts
+an async script if a signal is aborted.
+
+_Usage:_
+
+```javascript
+const { checkpoint } = require('@consento/promise/checkpoint')
+
+async function longRunning ({ signal }: {}) {
+  const cp = checkpoint(signal)
+  let result
+  for (const data of internalIterator()) {
+    result += await cp(data) // An AbortError will be thrown if the passed-in signal happens to be aborted.
+  }
+  return result
+}
+```
+
+#### `cleanupPromise()`
+
+Versatile custom Promise implementation that allows to execute an async
+cleanup operation after a promise is resolved or rejected.
+
+_Usage:_
+
+```javascript
+import { cleanupPromise } = require('@consento/promise/cleanupPromise')
+
+const result = await cleanupPromise((resolve, reject, signal) => {
+  setTimeout(resolve, Math.random() * 500) // Resolve & reject work like in regular promises
+  setTimeout(reject, Math.random() * 500)
+  const abortHandler = () => {
+    reject(new Error('aborted'))
+  }
+  signal.addEventListener('abort', abortHandler)
+  return () => {
+    // Executed after resolve or reject is called.
+    signal.removeEventListener('abort, abortHandler)
+  }
+})
+
+cleanupPromise(..., { timeout: 500, signal }) // You can also pass-in a parent signal or a timeout!
+```
+
+#### `isPromiseLike()`
+
+TypeScript util that helps to identify if a promise is a [`PromiseLike`][PromiseLike] instance.
+
+[PromiseLike]: https://github.com/microsoft/TypeScript/blob/1bd8e388aeda1df0f1dbc2a1a0ef9361a0d43d6f/src/lib/es5.d.ts#L1401-L1409
+
+#### `raceWithSignal()`
+
+Similar to `Promise.race` but aborts all promises that don't win the race.
+
+_Usage:_
+
+```javascript
+const { raceWithSignal } = require('@consento/promise/raceWithSignal')
+
+// Here the request to google will be aborted if the timeout of 500ms is reached.
+await raceWithSignal(signal => {
+  return [
+    fetch('http://google.com', { signal }),
+    new Promise((resolve, reject) => setTimeout(reject, 500, new Error('timeout')))
+  ]
+})
+
+await raceWithSignal(..., inputSignal) // You can also pass-in a signal that you maintain.
+```
+
+#### `toPromise()`
+
+For when you need to make sure that you have a promise at hand.
+
+Turns any given js object into a Promise. It will also cache the
+Promises to save memory and instantiation time.
+
+There are several legacy promise implementations that are similar
+but different from the `Promise` javascript standard, this method
+will turn those promises into actual Promises.
```

Comment:
Might be good to say what the main difference is from `Promise.resolve`, also how does the cache get invalidated / avoid memory leaks?

Reply:
... it does what `Promise.resolve` does ... will remove it.

Reply:
Removed in https://github.com/consento-org/promise/pull/1/commits/bbd8a77a4103050f839de2c82f8d2589ae0fe07c